src/analysis/retail/monk/windwalker/modules/apl/AplCheck.tsx

```diff
@@ -118,22 +151,188 @@ export const apl = build([
+        hasResource(RESOURCE_TYPES.ENERGY, {
+          // over 85% of max energy
+          atLeast:
+            (BASE_ENERGY_MAX +
+              (hasTalent(TALENTS.ASCENSION_TALENT) ? ASCENSION_ENERGY_MAX_ADDITION : 0) +
+              (hasTalent(TALENTS.INNER_PEACE_TALENT) ? INNER_PEACE_ENERGY_MAX_ADDITION : 0)) *
+            0.85,
+        }),
+        not(lastSpellCast(SPELLS.TIGER_PALM)),
+      ),
+      (tense) => (
+        <>
+          you have at most {hasTalent(TALENTS.ASCENSION_TALENT) ? 4 : 3}{' '}
+          <SpellLink spell={RESOURCE_TYPES.CHI} />, AND about to cap energy
+        </>
+      ),
+    ),
+  },
+  {
+    spell: SPELLS.SPINNING_CRANE_KICK,
+    condition: buffStacks(SPELLS.DANCE_OF_CHI_JI_BUFF, { atLeast: 2 }),
+  },
+  {
+    spell: TALENTS.WHIRLING_DRAGON_PUNCH_TALENT,
+    condition: describe(buffPresent(SPELLS.WHIRLING_DRAGON_PUNCH_USABLE), (tense) => (
+      <>
+        <SpellLink spell={TALENTS.RISING_SUN_KICK_TALENT} /> and{' '}
+        <SpellLink spell={TALENTS.FISTS_OF_FURY_TALENT} /> {tenseAlt(tense, 'are', 'were')} on
+        cooldown.
+      </>
+    )),
+  },
+  {
+    spell: SPELLS.CRACKLING_JADE_LIGHTNING,
+    condition: buffStacks(TALENTS.LAST_EMPERORS_CAPACITOR_TALENT, { atLeast: 20 }),
+  },
+  TALENTS.SLICING_WINDS_TALENT,
+  {
+    spell: TALENTS.FISTS_OF_FURY_TALENT,
+    condition: hasChi(3),
+  },
+  {
+    spell: TALENTS.RISING_SUN_KICK_TALENT,
+    condition: hasChi(2),
+  },
+  {
+    spell: SPELLS.BLACKOUT_KICK,
+    condition: describe(
+      and(
+        hasChi(1),
+        or(
+          and(
+            hasTalent(TALENTS.KNOWLEDGE_OF_THE_BROKEN_TEMPLE_TALENT),
+            buffStacks(SPELLS.TEACHINGS_OF_THE_MONASTERY, { atLeast: 8 }),
+          ),
+          and(
+            not(hasTalent(TALENTS.KNOWLEDGE_OF_THE_BROKEN_TEMPLE_TALENT)),
+            buffStacks(SPELLS.TEACHINGS_OF_THE_MONASTERY, { atLeast: 4 }),
+          ),
+        ),
+      ),
+      (tense) => (
+        <>
+          at max stacks of <SpellLink spell={TALENTS.TEACHINGS_OF_THE_MONASTERY_TALENT} />
+        </>
+      ),
+    ),
+  },
+  {
+    spell: SPELLS.SPINNING_CRANE_KICK,
+    condition: buffPresent(SPELLS.DANCE_OF_CHI_JI_BUFF),
+  },
+  {
+    spell: TALENTS.CHI_BURST_WINDWALKER_TALENT,
+    condition: buffPresent(SPELLS.CHI_BURST_PROC),
+  },
+  {
+    spell: SPELLS.CRACKLING_JADE_LIGHTNING,
+    condition: describe(
+      and(
+        hasTalent(TALENTS.LAST_EMPERORS_CAPACITOR_TALENT),
+        buffStacks(SPELLS.LAST_EMPERORS_CAPACITOR_BUFF, { atLeast: 15 }),
+      ),
+      (tense) => (
+        <>
+          <SpellLink spell={TALENTS.LAST_EMPERORS_CAPACITOR_TALENT} /> would overcap before next
+          good use
+        </>
+      ),
+    ),
+  },
+  {
+    spell: SPELLS.TIGER_PALM,
+    condition: describe(
+      and(
+        hasResource(RESOURCE_TYPES.ENERGY, {
+          atLeast: hasTalent(TALENTS.CELERITY_TALENT) ? 55 : 60,
+        }),
```

Comment:
this won't work correctly. you have a couple of options here.

1. have your apl built by a function that takes the `combatant` as a parameter, then use `combatant.hasTalent(TALENTS.CELERITY_TALENT)` to check this. like `apl = (combatant: Combatant) => build([...]);`
2. use a 2nd rule, one with the 55 and `hasTalent`, and the other with 60 and `not(hasTalent)`

the reason for this is that `condition.hasTalent(...)` returns a *condition object*, not true/false. the condition object is used to check the condition at every event.